src/vs/workbench/services/runtimeStartup/common/runtimeStartup.ts

```diff
@@ -344,11 +350,59 @@ export class RuntimeStartupService extends Disposable implements IRuntimeStartup
 	}
 
 	/**
-	 * Completes the language runtime discovery phase. If no runtimes were
-	 * started or will be started, automatically start one.
+	 * Signals that the runtime discovery phase is completed only after all
+	 * extension hosts have completed runtime discovery.
+	 *
+	 * If no runtimes were started or will be started, automatically start one.
+	 */
+	public completeDiscovery(id: number): void {
+		// Update the extension host's runtime discovery state to 'Complete'
+		this._extensionHostDiscoveryCompleteByMainThreadLanguageRuntimeId.set(id, true);
```

Comment:
For hygiene, I think we should reset all the values in this array to `false` when starting a new discovery pass. Currently discovery happens only once but that isn't a guarantee (there are a few open issues requesting the ability to refresh/re-discover for example)